python/pyarrow/tests/test_cpp_internals.py

```diff
@@ -31,3 +34,17 @@ def wrapper(case=case):
 
 
 inject_cpp_tests(globals())
+
+
+def test_pyarrow_include():
+    # We need to make sure that pyarrow/include is always
+    # created. Either with PyArrow C++ header files or with
+    # Arrow C++ and PyArrow C++ header files together
+
+    source = os.path.dirname(os.path.abspath(__file__))
+    pyarrow_dir = os.path.abspath(pjoin(source, '..'))
```

Comment:
I need to go from `test` folder to the parent `pyarrow` folder to test if the `include` folder is present there.

Reply:
Sorry... I'm not sure why `os.path.abspath()` is needed for it.
Could you show the followings in your environment?

* The `source` value
* `ls -lah ${source}` (`${source}` is the `source` value)
* The `pyarrow_dir` value
* `ls -lah ${pyarrow_dir}` (`${pyarrow_dir}` is the `pyarrow_dir` value)

Reply:
For me it seems to work also with a ".." in it:

```
In [9]: os.path.exists("/home/joris/scipy/repos/arrow/python/pyarrow/tests/../include/arrow/python/")
Out[9]: True
```

Reply:
But so about @kou's original comment, I think doing `os.path.dirname(source)` also works because if `source` is already a directory, `dirname` gives the parent directory. 
Now personally I find `..` more explicit to read that it goes one level up.

Reply:
Oh, sorry for my confusion.
Can change to `pyarrow_dir = os.path.dirname(source)`, @kou do you agree?

Reply:
Ah, now I see Joris you already made a suggestion. Will commit that 👍